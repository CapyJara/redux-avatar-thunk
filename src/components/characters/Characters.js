import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

function Characters({ characters }) {

  const liStyle = {
    padding: '10px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center'
  };

  const characterList = characters.map(character => {
    return (
      <li style={liStyle} key={character.name}>
        <Character character={character} />
      </li>
    );
  });

  const ulStyle = {
    display: 'flex',
    flexWrap: 'wrap'
  };

  return <ul style={ulStyle} >{characterList}</ul>;
}

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired
  ) 
};

export default Characters;
