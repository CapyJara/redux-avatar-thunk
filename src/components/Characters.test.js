import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component', () => {
  it('renders a Characters', () => {
    const characters = [
      {
        image: 'image@image.png',
        name: 'person person'
      },
      {
        image: 'holysmokes.png',
        name: 'yolo dude bro'
      },
      {
        image: 'neatPic.png',
        name: 'sassy queen'
      }
    ];

    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
