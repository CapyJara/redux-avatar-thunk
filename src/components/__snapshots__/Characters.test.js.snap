// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders a Characters 1`] = `
<ul>
  <li
    key="person person"
  >
    <Character
      character={
        Object {
          "image": "image@image.png",
          "name": "person person",
        }
      }
    />
  </li>
  <li
    key="yolo dude bro"
  >
    <Character
      character={
        Object {
          "image": "holysmokes.png",
          "name": "yolo dude bro",
        }
      }
    />
  </li>
  <li
    key="sassy queen"
  >
    <Character
      character={
        Object {
          "image": "neatPic.png",
          "name": "sassy queen",
        }
      }
    />
  </li>
</ul>
`;
